Sources/FoundationNetworking/URLSession/libcurl/MultiHandle.swift

```diff
@@ -48,6 +48,15 @@ extension URLSession {
         fileprivate var timeoutSource: _TimeoutSource? = nil
         private var reentrantInUpdateTimeoutTimer = false
         
+        // Only use serialization for OpenSSL < 1.1.0 which has race conditions during cleanup
+        private static let _needsCleanupSerialization: Bool = {
+            let version = CFURLSessionSSLVersionInfo()
+            return version.major < 1 || (version.major == 1 && version.minor < 1)
```

Comment:
Since curl can be built with TLS libraries other than OpenSSL, and those libraries would return `{0, 0, 0}` here, we'd be performing cleanup serialization that wouldn't have happened before/might not be needed. Would we want to add some way to check that SSL version parsing did succeed and is in fact OpenSSL before checking the version number?